packages/meditrak-server/src/routes/postChanges.js

```diff
@@ -64,8 +65,17 @@ export async function postChanges(req, res) {
     assertAnyPermissions([assertBESAdminAccess, surveyResponsePermissionsChecker]),
   );
 
-  for (const { action, translatedPayload } of translatedChanges) {
+  for (const { action, translatedPayload, ...rest } of translatedChanges) {
     await ACTION_HANDLERS[action](models, translatedPayload);
+
+    if (action === SUBMIT_SURVEY_RESPONSE) {
+      const { waitForAnalyticsRebuild } = rest;
+      if (waitForAnalyticsRebuild) {
+        const { database } = models;
+        await AnalyticsRefresher.executeRefresh(database);
```

Comment:
Just minor but perhaps we can do this outside of the for loop, just in case in the future there are multiple changes that need to refresh the analytics table. Just a thought so no need to change it now